db.js

```diff
@@ -213,8 +213,9 @@ exports.init = function (sbot, config) {
           for (var i = 0; i < msgVals.length; ++i) {
             if (i === msgVals.length - 1) {
               // last KVT, let's update the latest state
+              updateState({ key: keys[i], value: msgVals[i] })
               log.add(keys[i], msgVals[i], (err, kvt) => {
-                if (!err && kvt) updateState(kvt)
```

Comment:
It was changed to updateState first to match publish and addImmediately. I'll fix the error checking.

Reply:
Yeah, I see the matching, what I'm thinking is that we probably *should* in all cases call `updateState` after `log.add` completes successfully. Say there is an error writing to the log, can things go bad if the latest KVT is out of sync?

Reply:
Good question. The question is in what cases log.add would fail, like it could def. fail if you ran out of space. If you do updateState and then only if log.add works, update post, then EBT would be in a wrong state and try to get the same message that it can't add to the database because it thinks its there. I think the reason it was there before might have been to handle concurrent writes, but that is anyway fucked because validate is a async function. I'm leaning towards making it consistent with updating post & updateState at the same time.